scripts/konduit.sh

```diff
@@ -0,0 +1,223 @@
+   if ! kubectl -n "${NAMESPACE}" exec -i deployment/"${INSTANCE}" -- echo; then
+      exit
+      fi
+}
+
+SetPorts()
+{
+   # Get a random DEST port for the k8 container
+   # so there is minimal conflict between users
+   DEST_PORT=0
+   until [ $DEST_PORT -gt 1024 ]; do
+	   DEST_PORT=$RANDOM
+   done
+
+   # Get a random LOCAL port
+   # so we can have more than one session if wanted
+   LOCAL_PORT=0
+   until [ $LOCAL_PORT -gt 1024 ]; do
+	   LOCAL_PORT=$RANDOM
+	   nc -z 127.0.0.1 $LOCAL_PORT 2> /dev/null && LOCAL_PORT=0 # try again if it's in use
+   done
+}
+
+SetDBpsql()
+{
+   PORT=${DB_PORT}
+   # Get DB settings
+   # Either from the app DATABASE_URL or the AZURE KV secret
+   if [ "${DBName}" = "" ]; then
+      K8_URL=$(echo 'echo $DATABASE_URL' | kubectl -n "${NAMESPACE}" exec -i deployment/"${INSTANCE}" -- sh)
+      DB_URL=$(echo "${K8_URL}" | sed "s/@[^~]*\//@127.0.0.1:${LOCAL_PORT}\//g")
+      DB_NAME=$(echo "${K8_URL}" | awk -F"@" '{print $2}' | awk -F":" '{print $1}')
+   else
+      KV_URL=$(az keyvault secret show --name "${DBName}"-database-url --vault-name "${KV}" | jq -r .value)
+      DB_URL=$(echo "${KV_URL}" | sed "s/@[^~]*\//@127.0.0.1:${LOCAL_PORT}\//g")
+      DB_NAME=$(echo "${KV_URL}" | awk -F"@" '{print $2}' | awk -F":" '{print $1}')
+   fi
+}
+
+SetDBredis()
+{
+   PORT=${REDIS_PORT}
+   # Get DB settings
+   # Either from the app REDIS_URL or the AZURE KV secret
+
+   # Not tested from an azure backing service
+
+   if [ "${DBName}" = "" ]; then
+      K8_URL=$(echo 'echo $REDIS_URL' | kubectl -n "${NAMESPACE}" exec -i deployment/"${INSTANCE}" -- sh)
+      DB_URL=$(echo "$K8_URL" | sed "s/\/\/[^~]*/\/\/127.0.0.1:${LOCAL_PORT}\//g")
+      DB_NAME=$(echo "$K8_URL" | awk -F"/" '{print $3}' | awk -F":" '{print $1}')
+   else
+      KV_URL=$(az keyvault secret show --name "${DBName}"-database-url --vault-name "${KV}" | jq -r .value)
+      DB_URL=$(echo "${KV_URL}" | sed "s/\/\/[^~]*/\/\/127.0.0.1:${LOCAL_PORT}\//g")
+      DB_NAME="${DBName}"
+   fi
+}
+
+OpenTunnels()
+{
+   # Open netcat tunnel between k8 deployment and postgres database
+   # Timeout of 8 hours set for an interactive session
+   # Testing for kubectl deployment with multiple replicas always hit the same pod (the first one?),
+   # will have to revisit if it becomes an issue
+   echo 'nc -v -lk -p '${DEST_PORT}' -w '${TMOUT}' -e /usr/bin/nc -w '${TMOUT} "${DB_NAME}" "${PORT}" | kubectl -n "${NAMESPACE}" exec -i deployment/"${INSTANCE}" -- sh &
+
+   # Open local tunnel to k8 deployment
+   kubectl port-forward -n "${NAMESPACE}" deployment/"${INSTANCE}" ${LOCAL_PORT}:${DEST_PORT} &
+}
+
+RunPsql()
+{
+   if [ "$Inputfile" = "" ]; then
+		psql -d "$DB_URL" --no-password
+	elif [ "$CompressedInput" = "" ]; then
+		psql -d "$DB_URL" --no-password < "$Inputfile"
+	else
+		gzip -d --to-stdout "${Inputfile}" | psql -d "$DB_URL" --no-password
+	fi
+}
+
+RunPgdump()
+{
+   TMOUT=3600
```

Comment:
Maybe add a comment to increase if 1h is not enough

Reply:
Added comment and moved variable as it was in the wrong place